### PR TITLE
fix(web): support shapefile based on esri

### PR DESCRIPTION
**Pull request by pyshx**
# Overview
This PR adds support for Shapefile format in classic based on ERSI's white paper on [it](https://www.esri.com/content/dam/esrisites/sitecore-archive/Files/Pdfs/library/whitepapers/pdfs/shapefile.pdf).

### CODE REVIEW

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    const fileExt = key.slice(-4).toLowerCase();
+    const fileName = key.slice(0, -4);
+
+    if (fileExt === ".shp") {
+      names.push(fileName);
+      zip.files[fileName + ".shp"] = zip.files[key];
+    } else if (fileExt === ".prj") {
+      projections[fileName] = proj4(await zip.files[key].async("text"));
+    } else if (
+      fileExt === ".json" ||
```
Comment (keiya01) [inline web/src/classic/core/mantle/data/shapefile/parseZip.ts:45]: The length of `.json` is 5, so doesn't `fileExt` variable have the value?

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  const geojson = await Promise.all(
+    names.map(async name => {
+      const lastDotIdx = name.lastIndexOf(".");
+      if (lastDotIdx > -1 && name.slice(lastDotIdx).indexOf("json") > -1) {
+        const parsed = JSON.parse(await zip.files[name].async("text"));
+        parsed.fileName = name.slice(0, lastDotIdx);
+        return parsed;
+      } else if (whiteList && whiteList.indexOf(name.slice(lastDotIdx + 1)) > -1) {
```
Comment (keiya01) [inline web/src/classic/core/mantle/data/shapefile/parseZip.ts:65]:
```suggestion
      } else if (whiteList.indexOf(name.slice(lastDotIdx + 1)) > -1) {
```
Seems `whiteList` is always defined.

Comment (keiya01) [reply web/src/classic/core/mantle/data/shapefile/parseZip.ts:65]: BTW isn't `whiteList` used?

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  if (!names.length) {
+    throw new Error("no layers found");
+  }
+
+  const geojson = await Promise.all(
+    names.map(async name => {
+      const lastDotIdx = name.lastIndexOf(".");
+      if (lastDotIdx > -1 && name.slice(lastDotIdx).indexOf("json") > -1) {
+        const parsed = JSON.parse(await zip.files[name].async("text"));
+        parsed.fileName = name.slice(0, lastDotIdx);
```
Comment (keiya01) [inline web/src/classic/core/mantle/data/shapefile/parseZip.ts:63]: What's this for?
And also this `parsed.name` : https://github.com/reearth/reearth/pull/984/files#diff-ccc3ed5ab320705c56e3f70d6adcd405a91706c37013443505ea11c785208fbcR67

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts (hunk 1 above)
Comment (pyshx) [reply web/src/classic/core/mantle/data/shapefile/parseZip.ts]: Done.

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts (hunk 2 above)
Comment (pyshx) [reply web/src/classic/core/mantle/data/shapefile/parseZip.ts]: whiteList isn't always defined, it was optional parameter but I've removed as we aren't really using it...

File: web/src/classic/core/mantle/data/shapefile/parseZip.ts (hunk 3 above)
Comment (pyshx) [reply web/src/classic/core/mantle/data/shapefile/parseZip.ts]: Fixed.

**Review by airslice — APPROVED** [review]

**Pull request merged**